### DISCUSSION TITLE
Webhook for Product change is not working

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [Shopify/shopify-app-template-node#793] [Q&A] [answered]
I added the below code in index.js

Shopify.Webhooks.Registry.addHandler("PRODUCTS_UPDATE", {
  path: "/webhooks",
  webhookHandler: async (topic, shop, body) => {
    console.log("Product Updated");
  },
});

this webhook is not getting triggered when the product is updated. Am I missing anything here? How to make it work? I have permission to read product scope.

**Comment by dan-gamble** [comment, marked as answer]
You need to register the webhook too, ideally on auth.

```javascript
    const response = await Shopify.Webhooks.Registry.register({
      shop: session.shop,
      accessToken: session.accessToken as string,
      topic: 'PRODUCTS_UPDATE',
      path: '/webhooks',
    });
```

**Reply by ramkishore-yerubandi** [reply]
thanks. Its working with the change
